### ISSUE TITLE
Removing file from stage with git breaks Atom git-integration

### ISSUE THREAD

**Issue by Algram** [atom/atom#12420]
### Description
If you use `git reset HEAD <filename>` to remove a file from staging that has been modified in Atom the git information highlighting stops working. It doesn't change state and shows outdated git information.

### Steps to Reproduce

1. `File -> Add Project Folder` (needs to be a git-Project)
2. Add two Files `foo.js` and `bar.js` and make changes to both
3. Use your command line to execute `git add all` in your project directory
4. Remove one file from the staging area `git reset HEAD foo.js`
5. Go back into Atom, `foo.js` is still marked as staged

**Expected behavior:**
`foo.js` is marked as either newly-added or modified.

**Actual behavior:**
`foo.js` is marked as staged.

### Versions
```
Atom    : 1.9.5
Electron: 0.37.8
Chrome  : 49.0.2623.75
Node    : 5.10.0

apm  1.10.0
npm  2.15.8
node 4.4.5
python 2.7.12
git 2.7.4
```

**Comment by 50Wliu** [comment]
@Algram Please try updating to the latest Atom version, currently 1.9.8 and see if you can still reproduce an issue.  1.9.6 fixed a Git regression that might also fix your bug.

Also, in the future please entirely fill out the provided issue template.  If you had, you would have realized that you were not on the latest version.

**Comment by Algram** [comment]
Fixed in 1.9.8.

@50Wliu Sorry for not using the latest version. It is quite a hassle to update a package that is not updated on the repos you are using on Linux.
